packages/edit-site/src/components/sidebar-navigation-screen-navigation-menu/index.js

```diff
@@ -25,24 +27,163 @@ import {
 const noop = () => {};
 
 export default function SidebarNavigationScreenNavigationMenu() {
+	const {
+		deleteEntityRecord,
+		saveEntityRecord,
+		editEntityRecord,
+		saveEditedEntityRecord,
+	} = useDispatch( coreStore );
+
+	const { createSuccessNotice, createErrorNotice } =
+		useDispatch( noticesStore );
+
 	const postType = `wp_navigation`;
 	const {
+		goTo,
 		params: { postId },
 	} = useNavigator();
 
-	const { record: navigationMenu, isResolving: isLoading } = useEntityRecord(
+	const { record: navigationMenu, isResolving } = useEntityRecord(
 		'postType',
 		postType,
 		postId
 	);
 
+	const { getEditedEntityRecord, isSaving, isDeleting } = useSelect(
+		( select ) => {
+			const {
+				isSavingEntityRecord,
+				isDeletingEntityRecord,
+				getEditedEntityRecord: getEditedEntityRecordSelector,
+			} = select( coreStore );
+
+			return {
+				isSaving: isSavingEntityRecord( 'postType', postType, postId ),
+				isDeleting: isDeletingEntityRecord(
+					'postType',
+					postType,
+					postId
+				),
+				getEditedEntityRecord: getEditedEntityRecordSelector,
+			};
+		},
+		[ postId, postType ]
+	);
+
+	const isLoading = isResolving || isSaving || isDeleting;
+
 	const menuTitle = navigationMenu?.title?.rendered || navigationMenu?.slug;
 
+	const handleSave = async ( edits = {} ) => {
+		// Prepare for revert in case of error.
+		const originalRecord = getEditedEntityRecord(
+			'postType',
+			'wp_navigation',
+			postId
+		);
+
+		// Apply the edits.
+		editEntityRecord( 'postType', postType, postId, edits );
+
+		// Attempt to persist.
+		try {
+			await saveEditedEntityRecord( 'postType', postType, postId, {
+				throwOnError: true,
+			} );
+			createSuccessNotice( __( 'Renamed Navigation menu' ), {
+				type: 'snackbar',
+			} );
+		} catch ( error ) {
+			// Revert to original in case of error.
+			editEntityRecord( 'postType', postType, postId, originalRecord );
+
+			createErrorNotice(
+				sprintf(
+					/* translators: %s: error message describing why the navigation menu could not be renamed. */
+					__( `Unable to rename Navigation menu (%s).` ),
+					error?.message
+				),
+
+				{
+					type: 'snackbar',
+				}
+			);
+		}
+	};
+
+	const handleDelete = async () => {
+		try {
+			await deleteEntityRecord(
+				'postType',
+				postType,
+				postId,
+				{
+					force: true,
+				},
+				{
+					throwOnError: true,
+				}
+			);
+			createSuccessNotice( __( 'Deleted Navigation menu' ), {
+				type: 'snackbar',
+			} );
+			goTo( '/navigation' );
+		} catch ( error ) {
+			createErrorNotice(
+				sprintf(
+					/* translators: %s: error message describing why the navigation menu could not be deleted. */
+					__( `Unable to delete Navigation menu (%s).` ),
+					error?.message
+				),
+
+				{
+					type: 'snackbar',
+				}
+			);
+		}
+	};
+	const handleDuplicate = async () => {
+		try {
+			const savedRecord = await saveEntityRecord(
+				'postType',
+				postType,
+				{
+					title: menuTitle + __( ' (Copy)' ),
```

Comment:
I think this should use a sprintf so that translators can put (copy) before the navigation name if that's necessary in their language.

Reply:
✅